packer is not going to building using go1.2.1

I'm trying to build it on Ubuntu 14.04 Trusty, using `golang` from upstream repositories.

    $ dpkg -s golang | grep Version
    Version: 2:1.2.1-2ubuntu1

    $ go version
    go version go1.2.1 linux/amd64

    $ make updatedeps
    go get -u github.com/mitchellh/gox
    go get -u golang.org/x/tools/cmd/stringer
    go list ./... \
		| xargs go list -f '{{join .Deps "\n"}}' \
		| grep -v github.com/mitchellh/packer \
		| sort -u \
		| xargs go get -f -u -v
    flag provided but not defined: -f

When I changed `-f` to `-fix` I get another errors

    ...
    github.com/pierrec/lz4
    # github.com/pierrec/lz4
    ../../pierrec/lz4/lz4.go:86: undefined: sync.Pool
    ...
    github.com/rackspace/gophercloud/openstack
    # google.golang.org/cloud/compute/metadata
    ../../../google.golang.org/cloud/compute/metadata/metadata.go:54: unknown net.Dialer field 'KeepAlive' in struct literal
    ...
    make: *** [updatedeps] Error 123

It seems that dependencies will not compile under golang1.2 and we need to update documentation to reflect this change.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -53,8 +53,8 @@ it raises the chances we can quickly merge or address your contributions.
 If you have never worked with Go before, you will have to complete the
 following steps in order to be able to compile and test Packer.
 
-1. Install Go. Make sure the Go version is at least Go 1.2. Packer will not work with anything less than
-   Go 1.2. On a Mac, you can `brew install go` to install Go 1.2.
+1. Install Go. Make sure the Go version is at least Go 1.4. Packer will not work with anything less than
+   Go 1.4. On a Mac, you can `brew install go` to install Go 1.4.
 
 2. Set and export the `GOPATH` environment variable and update your `PATH`.
    For example, you can add to your `.bash_profile`.

